packages/shared/src/services/collections.ts

```diff
@@ -0,0 +1,64 @@
+import Keyv from 'keyv';
+import { Knex } from 'knex';
+import { Table } from 'knex-schema-inspector/dist/types/table';
+import {
+	Accountability,
+	Collection,
+	CollectionMeta,
+	Helpers,
+	MutationOptions,
+	RawField,
+	SchemaOverview,
+} from '../types';
+export declare type RawCollection = {
+	collection: string;
+	fields?: RawField[];
+	schema?: Partial<Table> | null;
+	meta?: Partial<CollectionMeta> | null;
+};
+export declare interface CollectionsService {
+	knex: Knex;
+	helpers: Helpers;
+	accountability: Accountability | null;
+	schemaInspector: any; // should be ReturnType<typeof SchemaInspector>; with SchemaInspector from '@directus/schema';
```

Comment:
Same problem here with `SchemaInspector` from `@directus/schema`
The dependence to  `@directus/schema` was also present in :
- [fields.ts](https://github.com/directus/directus/blob/d7f27830e43015ef111c297172ab46ebca29c97b/packages/shared/src/services/fields.ts)
- [relations.ts](https://github.com/directus/directus/blob/d7f27830e43015ef111c297172ab46ebca29c97b/packages/shared/src/services/relations.ts)